### PR TITLE
dd_test: Update the method of getting partitions

### PR DESCRIPTION
**Pull request by YongxueHong**
To get partitions in during time continuously since
it will spend some time in syncing the operation of
creating partitions.

ID: 1695549
Signed-off-by: Yongxue Hong <yhong@redhat.com>

### CODE REVIEW

**Comment by YongxueHong** [comment]
@ngu-niny @CongLi  @XueqiangWei  @lolyu  pls help me review thanks,

File: generic/tests/dd_test.py [hunk 1]
```diff
@@ -83,6 +90,7 @@ def _get_file(filename, select, test=test):
 
     dev_partitioned = []
     for arg in ['dd_if', 'dd_of']:
+        partitions = []
```
Comment (lolyu) [inline generic/tests/dd_test.py]: `partitions` is used only for the `if` block, no need to put it here.

File: generic/tests/dd_test.py [hunk 2]
```diff
@@ -62,6 +62,13 @@ def _get_file(filename, select, test=test):
                 logging.error(err)
                 test.error(err)
 
+    def _get_data_disk_partitions():
```
[the remark this answers is not in the document]
Comment (YongxueHong) [reply generic/tests/dd_test.py]: thanks, have updated.

File: generic/tests/dd_test.py (hunk 1 above)
Comment (YongxueHong) [reply generic/tests/dd_test.py]: thanks, have updated.

**Comment by CongLi** [comment]
Ack.
Finshed=1000, PASS=1000

File: generic/tests/dd_test.py [hunk 3]
```diff
@@ -62,6 +62,14 @@ def _get_file(filename, select, test=test):
                 logging.error(err)
                 test.error(err)
 
+    def _check_disk_partitions_number():
+        """ Check the data disk partitions number. """
+        for _ in range(len(partitions)):
+            partitions.pop()
```
Comment (hereischen) [inline generic/tests/dd_test.py]:
```python
del partitions[:]
```

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by XueqiangWei** [comment]
LGTM,
Acked-by xuwei@redhat.com

**Comment by ngu-niny** [comment]
Test PASS on PPC:
'05-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.dd_test.readwrite.zero2disk' Finshed 2019-04-11 03:56:33 2019-04-11 03:57:47 PASS 73.851017
'06-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.dd_test.readwrite.disk2null' Finshed 2019-04-11 03:57:47 2019-04-11 03:59:01 PASS 74.082825
'07-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.dd_test.readonly.zero2disk' Finshed 2019-04-11 03:59:01 2019-04-11 04:00:16 PASS 74.296558
'08-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.dd_test.readonly.disk2null' Finshed 2019-04-11 04:00:16 2019-04-11 04:01:31 PASS 74.795519
'09-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.dd_test.multiple.zero2disk' Finshed 2019-04-11 04:01:31 2019-04-11 04:03:14 PASS 103.322184
'10-Host_RHEL.m8.u0.qcow2.spapr_vscsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.dd_test.multiple.disk2null' Finshed 2019-04-11 04:03:14 2019-04-11 04:04:34 PASS 80.48312

Acked-by: Nini Gu <ngu@redhat.com>

**Comment by YongxueHong** [comment]
thanks all.

**Pull request merged**